repair/repair.cc

```diff
@@ -1397,29 +1401,39 @@ static int do_repair_start(seastar::sharded<database>& db, sstring keyspace,
     }
 
     std::vector<sstring> cfs;
+    std::vector<schema_ptr> schemas;
     if (options.column_families.size()) {
         cfs = options.column_families;
         for (auto& cf : cfs) {
             try {
-                db.local().find_column_family(keyspace, cf);
+                schemas.push_back(db.local().find_column_family(keyspace, cf).schema());
             } catch(...) {
                 throw std::runtime_error(format("No column family '{}' in keyspace '{}'", cf, keyspace));
             }
         }
     } else {
-        cfs = list_column_families(db.local(), keyspace);
+        std::tie(cfs, schemas) = list_column_families(db.local(), keyspace);
+    }
+    if (cfs.empty()) return id;
+    for (size_t idx = 1; idx < schemas.size(); ++idx) {
+        if (schemas[idx - 1]->get_partitioner() != schemas[idx]->get_partitioner()) {
+            throw std::runtime_error(
+                    format("All column families repaired together have to have the same partitioner. Different partitioners found: {} and {}",
+                        schemas[idx - 1]->get_partitioner(), schemas[idx]->get_partitioner()));
+        }
```

Comment:
Better mention the table names too.

Reply:
s/column families/ tables.

Reply:
With this patch,  `nodetool repair keyspace` will fail if tables with different partitioner are inside this keyspace. In this case, user has to manually specify the tables and repair table by table. What if there are 1000 tables. 

I think this is a regression.

Reply:
This is true @asias good thing is that by default all the tables use default partitioner so the problem does not exist. It's on the user to not create table with custom partitioner if they find it problematic to repair them by manually selecting tables. I don't think it's a regression. It does not affect any of the existing keyspaces. Only when user decides to use the new functionality of per table partitioner then they will face the problem but that's part of the price for the feature. No feature use - no cost. Zero cost abstraction :)

Reply:
Is the partitioner even something that can be set by the user (via CQL). I didn't see any code that would persist and load the partitioner into the system schema tables.

Also, supporting this would open a whole can of worms, as a schema later can now possibly change token ownership (even if only within a single node).

Reply:
It is not possible at the moment. Initial user will be internal (CDC). We may later want to allow users to set per table partitioners but only during create table and disallow changing it using alter table.

Reply:
I changed the column families to tables and added table names.